src/jit/morph.cpp

```diff
@@ -18416,6 +18417,29 @@ class LocalAddressVisitor final : public GenTreeVisitor<LocalAddressVisitor>
         m_compiler->fgMorphLocalField(node, user);
         INDEBUG(m_stmtModified |= node->OperIs(GT_LCL_VAR);)
     }
+
+    //------------------------------------------------------------------------
+    // UpdateImplicitByRefCounter: updates the ref count for implicit byref params.
+    //
+    // Arguments:
+    //    lclNum - the local number to update the count for.
+    //
+    // Notes:
+    //    fgMakeOutgoingStructArgCopy checks the ref counts for implicit byref params when deciding
+    //    if it's legal to elide certain copies of them;
+    //    fgRetypeImplicitByRefArgs checks the ref counts when decides to undo promotions.
```

Comment:
when "it" decides?

Reply:
It happens here https://github.com/dotnet/coreclr/blob/4410262505b5ebe308476116e9e833b210394638/src/jit/morph.cpp#L5018:
```
        if (lvaIsImplicitByRefLocal(varNum))
        {
            LclVarDsc* varDsc = &lvaTable[varNum];
            // JIT_TailCall helper has an implicit assumption that all tail call arguments live
            // on the caller's frame. If an argument lives on the caller caller's frame, it may get
            // overwritten if that frame is reused for the tail call. Therefore, we should always copy
            // struct parameters if they are passed as arguments to a tail call.
            if (!call->IsTailCallViaHelper() && (varDsc->lvRefCnt(RCS_EARLY) == 1) && !fgMightHaveLoop())
            {
                varDsc->setLvRefCnt(0, RCS_EARLY);
                args->gtOp.gtOp1 = lcl;
                argEntry->node   = lcl;

                JITDUMP("did not have to make outgoing copy for V%2d", varNum);
                return;
            }
        }
```
(suddenly GitHub sttoped to show code from permalinks).

it happens during `fgMorphArgs` under such condition:
```
        if (copyBlkClass != NO_CLASS_HANDLE)
        {
            fgMakeOutgoingStructArgCopy(call, args, argIndex, copyBlkClass);
        }
```

Reply:
Sorry for not being clearer.

I was only suggesting you add the word "it" to the comment ... "when decides" lacks a subject, "when it decides" reads better.